### ISSUE TITLE
Confusion about the hyperparameters

### ISSUE THREAD

**Issue by xxdznl** [haozzh/FedCR#3]
Hi! @haozzh 
I couldn't reproduce the result of your paper and even got an unreasonable result when trying to reproduce FedAvg. 
“[2023-08-28 11:38:29,613][INFO] Epoch:[498]	lr =	0.01000	loss=	3.92252	acc_test=	23.05556
[2023-08-28 11:38:55,656][INFO] Epoch:[499]	lr =	0.01000	loss=	4.07925	acc_test=	22.43333
[2023-08-28 11:42:21,249][INFO] Final Epoch:[500]	lr =	0.01000	loss=	3.78364	acc_test=	24.40000
[2023-08-28 11:42:21,249][INFO] Final FT Epoch:[500]	lr =	0.01000	loss=	0.95096	_**acc_test=	75.75000**_”
So I am very confuseing about what the exact values of hyperparameters are.
Could you please relase a more specific list of the experimental details？

**Comment by haozzh** [comment]
Hello,may I ask what dataset you are based on? Using CIFAR-10 as an illustrative example, we need to configure a specific set of hyperparameters as follows to reproduce the mentioned accuracy.


_[2022-11-28 17:05:46,941][INFO] --------args----------
[2022-11-28 17:05:46,943][INFO] epochs: 500
[2022-11-28 17:05:46,943][INFO] test_freq: 1
[2022-11-28 17:05:46,943][INFO] num_users: 100
[2022-11-28 17:05:46,943][INFO] frac: 0.1
[2022-11-28 17:05:46,943][INFO] local_ep: 10
[2022-11-28 17:05:46,943][INFO] last_local_ep: 10
[2022-11-28 17:05:46,943][INFO] local_rep_ep: 10
[2022-11-28 17:05:46,943][INFO] local_bs: 48
[2022-11-28 17:05:46,944][INFO] bs: 10
[2022-11-28 17:05:46,944][INFO] lr: 0.01
[2022-11-28 17:05:46,944][INFO] globallr: 1
[2022-11-28 17:05:46,944][INFO] momentum: 0
[2022-11-28 17:05:46,944][INFO] weigh_delay: 0
[2022-11-28 17:05:46,944][INFO] mu: 0
[2022-11-28 17:05:46,944][INFO] lr_decay: 1.0
[2022-11-28 17:05:46,944][INFO] alpha: 0
[2022-11-28 17:05:46,944][INFO] beta: 0.001
[2022-11-28 17:05:46,944][INFO] beta2: 0
[2022-11-28 17:05:46,944][INFO] dimZ: 256
[2022-11-28 17:05:46,944][INFO] num_avg_train: 18
[2022-11-28 17:05:46,945][INFO] num_avg: 18
[2022-11-28 17:05:46,945][INFO] filepath: CIFAR10_fedavg_0.01_1_dir5_ep10.txt
[2022-11-28 17:05:46,945][INFO] method: fedavg
[2022-11-28 17:05:46,945][INFO] dataset: CIFAR10
[2022-11-28 17:05:46,945][INFO] frac_data: 0.7
[2022-11-28 17:05:46,945][INFO] rule: Dirichlet
[2022-11-28 17:05:46,945][INFO] class_main: 0
[2022-11-28 17:05:46,945][INFO] dir_a: 0.5
[2022-11-28 17:05:46,945][INFO] gpu: 0
[2022-11-28 17:05:46,945][INFO] seed: 23
[2022-11-28 17:05:46,946][INFO] device: cuda:0_


**And it converges to**


[2022-11-28 20:45:52,888][INFO] Epoch:[498]	lr =	0.01000	loss=	0.90558	acc_test=	75.38333
[2022-11-28 20:46:18,704][INFO] Epoch:[499]	lr =	0.01000	loss=	0.87784	acc_test=	75.82222
[2022-11-28 20:49:39,815][INFO] Final Epoch:[500]	lr =	0.01000	loss=	0.84143	acc_test=	76.34444
[2022-11-28 20:49:39,816][INFO] Final FT Epoch:[500]	lr =	0.01000	loss=	0.58217	acc_test=	83.65556

**Comment by haozzh** [comment]
Upon examining your results, despite not utilizing the optimal hyperparameters, the final stage exhibited a remarkable increase in accuracy ather the fine-tuning process, surging from an initial 24% to a 75%. This outcome didn't come as a surprise, given my own experimentation with fine-tuning, where I found it to be notably potent for personalized federated learning. This robust performance is likely the reason behind the impressive efficacy of the FedAvg-FT, compared to FedAvg.

**Comment by xxdznl** [comment]
thank you, it make sense.

> Upon examining your results, despite not utilizing the optimal hyperparameters, the final stage exhibited a remarkable increase in accuracy ather the fine-tuning process, surging from an initial 24% to a 75%. This outcome didn't come as a surprise, given my own experimentation with fine-tuning, where I found it to be notably potent for personalized federated learning. This robust performance is likely the reason behind the impressive efficacy of the FedAvg-FT, compared to FedAvg.

**Comment by xxdznl** [comment]
**noniid setting** : noniid, **dataset**: cifar-100 ,but the **class_main** was set to 5, when i set it to 15.I get the same result.
[2023-08-29 11:38:17,013][INFO] Epoch:[499]	lr =	0.05000	loss=	4.89818	acc_test=	28.94444
[2023-08-29 11:41:31,010][INFO] Final Epoch:[500]	lr =	0.05000	loss=	4.73088	acc_test=	29.80000
[2023-08-29 11:41:31,011][INFO] Final FT Epoch:[500]	lr =	0.05000	loss=	2.82783	acc_test=	54.42778

> Hello,may I ask what dataset you are based on? Using CIFAR-10 as an illustrative example, we need to configure a specific set of hyperparameters as follows to reproduce the mentioned accuracy.
> 
> _[2022-11-28 17:05:46,941][INFO] --------args---------- [2022-11-28 17:05:46,943][INFO] epochs: 500 [2022-11-28 17:05:46,943][INFO] test_freq: 1 [2022-11-28 17:05:46,943][INFO] num_users: 100 [2022-11-28 17:05:46,943][INFO] frac: 0.1 [2022-11-28 17:05:46,943][INFO] local_ep: 10 [2022-11-28 17:05:46,943][INFO] last_local_ep: 10 [2022-11-28 17:05:46,943][INFO] local_rep_ep: 10 [2022-11-28 17:05:46,943][INFO] local_bs: 48 [2022-11-28 17:05:46,944][INFO] bs: 10 [2022-11-28 17:05:46,944][INFO] lr: 0.01 [2022-11-28 17:05:46,944][INFO] globallr: 1 [2022-11-28 17:05:46,944][INFO] momentum: 0 [2022-11-28 17:05:46,944][INFO] weigh_delay: 0 [2022-11-28 17:05:46,944][INFO] mu: 0 [2022-11-28 17:05:46,944][INFO] lr_decay: 1.0 [2022-11-28 17:05:46,944][INFO] alpha: 0 [2022-11-28 17:05:46,944][INFO] beta: 0.001 [2022-11-28 17:05:46,944][INFO] beta2: 0 [2022-11-28 17:05:46,944][INFO] dimZ: 256 [2022-11-28 17:05:46,944][INFO] num_avg_train: 18 [2022-11-28 17:05:46,945][INFO] num_avg: 18 [2022-11-28 17:05:46,945][INFO] filepath: CIFAR10_fedavg_0.01_1_dir5_ep10.txt [2022-11-28 17:05:46,945][INFO] method: fedavg [2022-11-28 17:05:46,945][INFO] dataset: CIFAR10 [2022-11-28 17:05:46,945][INFO] frac_data: 0.7 [2022-11-28 17:05:46,945][INFO] rule: Dirichlet [2022-11-28 17:05:46,945][INFO] class_main: 0 [2022-11-28 17:05:46,945][INFO] dir_a: 0.5 [2022-11-28 17:05:46,945][INFO] gpu: 0 [2022-11-28 17:05:46,945][INFO] seed: 23 [2022-11-28 17:05:46,946][INFO] device: cuda:0_
> 
> **And it converges to**
> 
> [2022-11-28 20:45:52,888][INFO] Epoch:[498] lr = 0.01000 loss= 0.90558 acc_test= 75.38333 [2022-11-28 20:46:18,704][INFO] Epoch:[499] lr = 0.01000 loss= 0.87784 acc_test= 75.82222 [2022-11-28 20:49:39,815][INFO] Final Epoch:[500] lr = 0.01000 loss= 0.84143 acc_test= 76.34444 [2022-11-28 20:49:39,816][INFO] Final FT Epoch:[500] lr = 0.01000 loss= 0.58217 acc_test= 83.65556
